pkg/authenticator/patcher.go

```diff
@@ -0,0 +1,69 @@
+package authenticator
+
+import (
+	"fmt"
+	"os"
+
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+const KubeConfigStubBinaryName = "server"
+
+func CreateKubeConfigPatcher() *KubeConfigPatcher {
+	return &KubeConfigPatcher{}
+}
+
+type KubeConfigPatcher struct{}
+
+func getKubeClientConfig(kubeConfig string) clientcmd.ClientConfig {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	loadingRules.ExplicitPath = kubeConfig
+
+	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &clientcmd.ConfigOverrides{})
+}
+
+// GenerateTempKubeConfigStubFile go through the kubeconfig file and replace all users using the Exec auth method by
+// an invocation of the stub binary.
+func (k *KubeConfigPatcher) GenerateTempKubeConfigStubFile(kubeConfig string) (string, error) {
```

Comment:
What does the `KubeConfigPatcher` struct bring into the equation? Feels like this could be a plain function without a message receiver.

Reply:
As usual, testing :o) But I can simplify it yeah since I don't plan to test it considering it's low value.